### ISSUE TITLE
IO Door sensor 

### ISSUE THREAD

**Issue by Daniels1959** [AdyRock/com.somfy.tahoma#172]
Afther the last update Somfy Tahoma 26.03.2024, the IO devices # Door  / window contact # are not responding to Homey

**Comment by AdyRock** [comment]
Can you enable the Information Log in the App settings screen, trigger one of the sensors and then send the log.

**Comment by Daniels1959** [comment]
OK Adrian,  just sended

**Comment by AdyRock** [comment]
Can you see any Tokens in the Local Tokens tab of the app settings?

**Comment by Daniels1959** [comment]
Error: No Somfy bridges have been detected.
Make sure the Developer mode has been enabled on you Somfy account.
The Somfy bridge should broadcast it's IP and PIN via mDNS once the option is enabled.

**Comment by Daniels1959** [comment]
![Token](https://github.com/AdyRock/com.somfy.tahoma/assets/149077248/f2d7d138-a539-4f0f-8c0f-8e0c40ad346a)

**Comment by AdyRock** [comment]
Have you got the Developer mode enabled on your account?

**Comment by Daniels1959** [comment]
no

**Comment by AdyRock** [comment]
OK, so that explains why you have no local tokens.
I would recommend you do that as the app will then be able to access your Tahoma directly instead of going via the cloud. It will be more reliable and faster (update every 3 seconds instead of 30 seconds).

However I am working on a fix for the issue caused by the update.

**Comment by Daniels1959** [comment]
OK , how to do this ?

**Comment by AdyRock** [comment]
Instructions are in the first post here under Local Access: https://community.homey.app/t/app-cloud-pro-somfy-tahoma-connexoon-v4-0-37-test-v4-0-75/35206

**Comment by Daniels1959** [comment]
ok Done , great results !!

**Comment by Daniels1959** [comment]
everything works correctly, great service Adrian, thank you

**Comment by AdyRock** [comment]
The update is also available that fixes the issue you had, but is not relevant now .

**Comment by Daniels1959** [comment]
may ask you an other question , related Senario player ?

**Comment by AdyRock** [comment]
Yes, fire away.

**Comment by Daniels1959** [comment]
a player retains his last position, so the status, for example: under, is retained until you change the status!  = like a switch !
Why not as a button ?

**Comment by Daniels1959** [comment]
![Player](https://github.com/AdyRock/com.somfy.tahoma/assets/149077248/709a5154-d54a-445d-9201-435ea9ed9eaf)

**Comment by AdyRock** [comment]
Sorry, I don't understand.
It looks like the capability has 4 states, so how would that be a button?
Maybe you could translate the text to see if it makes more sense to me.

**Comment by Daniels1959** [comment]
now always 1 status is always 1, and it also remains 1 if you don't press
with a button the status is 1 as long as you press it.
e.g.: I switch on a lamp with the player (ON). I switch that lamp off with another device. If I now want to switch this lamp on again with the player, I first have to change the status of the player, and then back to ON, to switch the lamp on.

**Comment by AdyRock** [comment]
I think I understand what you mean. I have order a controller so I can have a play.

**Comment by Daniels1959** [comment]
good morning Adrian , 
backt to the first issue , the issue came again , the IO devices # Door / window contact # are not responding to Homey
![token 28 03](https://github.com/AdyRock/com.somfy.tahoma/assets/149077248/9c0f96ab-ec09-46ae-815f-15b73144dea4)

**Comment by Daniels1959** [comment]
I just sended the Information Log

**Comment by Daniels1959** [comment]
Hi Adrian , related the Senario player , did you have a play with the controller ?

**Comment by AdyRock** [comment]
Sorry, yes. can you try the new test version of the app. This resets the state to null after 500ms so it can be changed again.
However, there seems to be an issue with the local events (which I have reported to Somfy) where it will only notify of different button press once. So, if you press Up then it won't notify of another Up press until the Down is pressed.
The cloud events work fine and it will send a notification for each press, so you can get Up, Up, Up, etc.

**Comment by Daniels1959** [comment]
hi Adrian , just did the test , seems not to work here , . i need still to go UP again to activate the Down commando

**Comment by AdyRock** [comment]
Yep, as I say, if you are using local mode, the events are on sent in an alternating sequence. So you can't respond to 2 or more consecutive presses of the same button. I have reported it to Somfy.
When I tested it in cloud mode, all the notifications came in, but in one notification, so there was up to 30 second delay and then just a ms or two between processing each event.
Hopefully, Somfy will fix it, but that could take a  long time.

**Comment by Daniels1959** [comment]
Hi Andy , did you have any response , related the report to Somfy ?

**Comment by AdyRock** [comment]
I had a response a few days ago, and I need to provide some examples of JSON responses. Unfortunately, I packed away my remote when I started my office refurbishment, so I don't have it to hand to generate the logs. I'm in the process of unpacking all the boxes now, so will get on it as soon as I come across the remote.

**Comment by Daniels1959** [comment]
OK  Andy

**Comment by Daniels1959** [comment]
Hi Adrian , i am still struggling with the scenario player, is it not possible to make a card, like the Philips Hue dimmer switch? (IF you press the button: up - Stop - Down) ? Now its the card : IF the external status is changed .

https://homey.app/nl-be/app/nl.philips.hue/Philips-Hue/

**Comment by AdyRock** [comment]
Sorry you lost me on this one. 

I have sent logs to Somfy, and I'm waiting for a response.
My Tahoma did update this afternoon, but I have been able to check if it fixes the issue.

**Comment by Daniels1959** [comment]
Hi , My Tahoma did also the update, I have tested the player , but still the same issue

**Comment by AdyRock** [comment]
> Hi Adrian , i am still struggling with the scenario player, is it not possible to make a card, like the Philips Hue dimmer switch? (IF you press the button: up - Stop - Down) ? Now its the card : IF the external status is changed .
> 
> https://homey.app/nl-be/app/nl.philips.hue/Philips-Hue/

You mean this one?
<img width="253" alt="image" src="https://github.com/AdyRock/com.somfy.tahoma/assets/38184396/7cd442d7-ba0a-486e-9e03-7e489256a967">

**Comment by Daniels1959** [comment]
![2024-06-21](https://github.com/AdyRock/com.somfy.tahoma/assets/149077248/6be11464-6e27-4102-b19a-9043a942682f)

**Comment by AdyRock** [comment]
Sorry, I'm still confused as the card I showed is the equivalent card of the Hue one you show.

**Comment by Daniels1959** [comment]
Yes but on the Hue dimmer is it a push function , related to the Player it is a Set function.
Means if you push a buton on the Hue , he has the status 1  ,  release the buton status 0 . Push again on the same buton Status  1 , release the buton status 0, en so on . 
On the Player , if you push the buton  ( exmpl. UP ) , he go to Status 1 and stay at status  1  , until you push a other buton ( STOP or Down)

**Comment by AdyRock** [comment]
Ah, I see. The Somfy remote doesn't provide a button-up event, but I can set it to null after a short delay so you get a separate event for each press (once they fix the bug).

**Comment by Daniels1959** [comment]
ok Adrian , please keep me informed when the bug is fixed
